sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/serializer/GeneratedProtobufSerializer.scala

```diff
@@ -14,60 +14,64 @@
  * limitations under the License.
  */
 
-package kalix.serializer
+package kalix.springsdk.impl.serializer
 
 import scala.jdk.CollectionConverters.CollectionHasAsScala
 import scala.jdk.CollectionConverters.MapHasAsJava
 
 import com.fasterxml.jackson.dataformat.protobuf.ProtobufMapper
 import com.google.protobuf.ByteString
 import com.google.protobuf.Descriptors.FileDescriptor
+import com.google.protobuf.UnsafeByteOperations
 import com.google.protobuf.{ Any => JavaPbAny }
+import kalix.javasdk.impl.ByteStringEncoding
+import kalix.serializer.Serializer
+import kalix.springsdk.impl.serializer.GeneratedProtobufSerializer.KalixGeneratedProto
 
-object JacksonProtobufSerializer {
+object GeneratedProtobufSerializer {
 
-  final val KalixJacksonProto = "jackson.proto.kalix"
+  final val KalixGeneratedProto = "proto.kalix.io/"
+
+  private val mapper = new ProtobufMapper()
 
   def buildSerializers(classLoader: ClassLoader, descriptor: FileDescriptor): Map[Class[_], Serializer] = {
 
-    val mapper = new ProtobufMapper()
     descriptor.getMessageTypes.asScala.map { messageType =>
-
       val packageName = descriptor.getFile.getPackage
-
       val className = packageName + "." + messageType.getName
       val classType = classLoader.loadClass(className)
 
-      val serializer = new Serializer {
-
-        val schemaWrapper = mapper.generateSchemaFor(classType)
+      val serializer = new GeneratedProtobufSerializer(mapper, classType)
+      (classType, serializer)
+    }.toMap
 
-        override def serialize(any: Any): JavaPbAny = {
-          val bytes =
-            mapper
-              .writer(schemaWrapper)
-              .writeValueAsBytes(any)
+  }
 
-          JavaPbAny
-            .newBuilder()
-            .setTypeUrl(s"${KalixJacksonProto}/${classType.getName}")
-            .setValue(ByteString.copyFrom(bytes))
-            .build()
-        }
+  def buildSerializersJava(classLoader: ClassLoader, descriptor: FileDescriptor): java.util.Map[Class[_], Serializer] =
+    buildSerializers(classLoader, descriptor).asJava
+}
 
-        override def deserialize(bytes: ByteString): Any =
-          mapper
-            .readerFor(classType)
-            .`with`(schemaWrapper)
-            .readValue(bytes.toByteArray)
-      }
+class GeneratedProtobufSerializer(mapper: ProtobufMapper, classType: Class[_]) extends Serializer {
 
-      (classType, serializer)
+  private val schemaWrapper = mapper.generateSchemaFor(classType)
 
-    }.toMap
+  override def serialize(any: Any): JavaPbAny = {
+    val bytes =
+      UnsafeByteOperations.unsafeWrap(
+        mapper
+          .writer(schemaWrapper)
+          .writeValueAsBytes(any))
 
+    JavaPbAny
+      .newBuilder()
+      .setTypeUrl(s"${KalixGeneratedProto}${classType.getName}")
+      .setValue(bytes)
```

Comment:
Ok, so not sure why we do the tag + bytes for our other custom type_url types but maybe there is a good reason for it vs just the bytes like here when it is not protobuf serialized to bytes?